### PR TITLE
Blocks: update warning style

### PR DESCRIPTION
**Pull request by johngodley**
## Description
Update the block warning style, moving the action buttons inline with the text, and adding a grey scrim. This prepares the warning component for the addition of a dropdown options menu. Discussion and reasons behind these changes can be found in #7604

vlidate

The block warning is shown in two situations:
- A block crashes and a `<BlockCrashWarning />` component is shown
- A block contains unexpected content and a `<BlockInvalidWarning />` is shown

In addition to style changes some text has been changed:
- The `Edit as HTML` button is now `Keep as HTML` to better reflect the action
- The message `This block appears to have been modified externally.` is now `This block has been modified externally.` to remove general vagueness

These items of text are likely to change in the future as the validation experience is improved, but for now should help with confusion.

Also to note that `InvalidBlockWarning` has been renamed to `BlockInvalidWarning` so it matches the other block components.

## How has this been tested?
Existing unit tests cover the changes to the HTML.

To manually test the block validation message:
- Create a paragraph
- Edit as HTML and remove the trailing `</p>` to trigger the block validation message
- Verify that the message changes from:
another_post___ _wordpress_latest_ _wordpress
to:
vlidate
- The buttons are right aligned and a grey scrim overlays the block content
- Verify that as the page width decreases the warning adjusts the buttons beneath the text (still left aligned):
another_post___ _wordpress_latest_ _wordpress
- Verify that the buttons continue to work as before

To test the block crash warning message:
- From a developer console type `wp.blocks.registerBlockType( 'myplugin/mr-crashy', { title: 'Mr. Crashy', category: 'common', icon: 'warning', edit() { throw new Error(); }, save() { return ''; } } );`
- Add `Mr Crashy` block to a post
- Confirm that the crash warning has changed from:
another_post___ _wordpress_latest_ _wordpress

To:

another_post___ _wordpress_latest_ _wordpress
- The warning now matches the width of a block

## Types of changes
This is a non-breaking change. The block warning functionality doesn't change.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by karmatosed** [comment]
Can I ask from a design perspective what happens with translations causing this to be on 2 lines? It seems it would throw everything off quite a lot. Also what happens on smaller screens?

**Review by karmatosed — APPROVED** [review]
Approving based on design. I do wonder if the fact the buttons move is great for consistency but we can explore other solutions once in. Having a giant modal isn't great either so this at least solves that one.

**Comment by johngodley** [comment]
@aduth you asked some questions in #7886 regarding the changes here. Does this PR answer them?

Specifically:

> increasing the z-index for the block warning

Joen adjusted the z-index for all block warnings. As far as I'm aware (and admittedly I know very little of Gutenberg at this stage), there's only an invalid block and crashed block warning. If that's the case then it shouldn't have any other effect, although I'm basing that on Joen's knowledge.

> Why was this moved?

As mentioned in f04f95a2fb7629c62b7f1a4003b69843fa587b61 it was moved to bring the crash warning to the same level in the DOM as the invalid block warning, allowing the style changes to apply to them both. Previously it was a level higher.

**Comment by aduth** [comment]
>Joen adjusted the z-index for all block warnings. As far as I'm aware (and admittedly I know very little of Gutenberg at this stage), there's only an invalid block and crashed block warning. If that's the case then it shouldn't have any other effect, although I'm basing that on Joen's knowledge.

The original worry was that we tend to think of only these ones as the only instances, but it was rewritten at one point to intentionally be generic, so we should be conscious of unintended impacts (including for future usage). Since it's increasing a `z-index` (and only slightly so), I doubt it's going to break much, but I'm still curious why it's necessary.

>As mentioned in f04f95a it was moved to bring the crash warning to the same level in the DOM as the invalid block warning, allowing the style changes to apply to them both. Previously it was a level higher.

This seems okay. Which styles specifically, though?

**Comment by johngodley** [comment]
> I'm still curious why it's necessary

I think Joen will need to answer that, although he is away so it will be a while. If this is something that you'd rather wait for then that's fine. If it makes sense to merge now and adjust later with Joen's info  then 👍 

> Which styles specifically, though?

`.editor-block-list__block .editor-warning`. I'm sure it would be possible to get the style looking the same without this DOM change, but this has simpler CSS and puts the warnings at the same place and using the same style so seems like a sensible thing, and should help for any future warnings.

**Comment by karmatosed** [comment]
I would +1 to merging it now and iterating.

**Comment by aduth** [comment]
If there are follow-up items, we should create an (ideally self-assigned) issue.

Otherwise, not a blocker for me.

File: editor/components/warning/style.scss [hunk 1]
```diff
@@ -1,27 +1,53 @@
-.editor-warning {
+.editor-block-list__block .editor-warning {
```
Comment (aduth) [inline editor/components/warning/style.scss]: I mentioned at https://github.com/WordPress/gutenberg/pull/7886#discussion_r202762204 that this component is used for more than just the warning for the single block in a list. It's also used for the top-level error handler, which this change broke. See #8677.

**Pull request merged**